lib/clock.ex

```diff
@@ -1,128 +1,58 @@
 defmodule UUIDv7.Clock do
   @moduledoc false
-  use GenServer
 
-  @type timestamp :: integer()
-  @type counter :: <<_::18>>
-  @type counter_seed :: <<_::17>>
-
-  @default_table_opts [:named_table, :public, :set]
-  @default_cleanup_interval_ms :timer.seconds(2)
-  @default_cleanup_tick_cutoff 2
-
-  # The threshold number where the counter will roll over.
-  @max_counter 2 ** 18
-
-  @compile {:inline, [update_counter: 2]}
-
-  @doc """
-  Get a unix millisecond timestamp and a clock sequence that fits in 18 bits.
-  """
-  @spec next(counter_seed()) :: {timestamp(), counter()}
-  def next(<<seed::17>>) do
-    timestamp_ref = :persistent_term.get(__MODULE__)
-    previous_ts = :atomics.get(timestamp_ref, 1)
-    current_ts = System.system_time(:millisecond)
-
-    # Time leap backwards protection.
-    current_ts = if current_ts < previous_ts, do: previous_ts, else: current_ts
-
-    :atomics.put(timestamp_ref, 1, current_ts)
+  # For macOS (Darwin) or Windows use 10 bits.
+  # Otherwise, use 12 bits (e.g. on Linux).
+  step_bits =
+    case :os.type() do
+      {:unix, :darwin} -> 10
+      {:win32, _} -> 10
+      {_, _} -> 12
+    end
 
-    # Rollover protection.
-    # If the counter is over the allotted bits, then we update the timestamp
-    # by 1 millisecond to preserve order, which will also reset the counter
-    # using the provided seed value.
-    clock =
-      with @max_counter <- update_counter(current_ts, seed) do
-        next_ts = current_ts + 1
-        :atomics.put(timestamp_ref, 1, next_ts)
-        update_counter(next_ts, seed)
-      end
+  ns_per_ms = 1_000_000
 
-    {current_ts, <<clock::big-unsigned-18>>}
-  end
+  @minimal_step_ns div(ns_per_ms, Bitwise.bsl(1, step_bits)) + 1
```

Comment:
Why not just set it to always be 1? I think it would be cleaner and still technically correct.

Reply:
I was trying to match the Postgres implementation as close as possible, but maybe I will have to compromise here

Reply:
I was trying to match [the Postgres implementation](https://github.com/postgres/postgres/blob/master/src/backend/utils/adt/uuid.c#L54) as close as possible, but maybe I will have to compromise here

Reply:
I know, but I do not see a point. If you do not have nanosecond precision there, then adding `1` is IMHO more secure than jumping somewhere wrt. precision of the OS. It is also secure to the same degree IMHO. This would also remove the cross-compilation problem.

I think it also reduces the potential problem of leaking information about runtime OS (if that is an issue). Theoretically it can leaks how many incoming requests tere are, but IMHO that is no different from "old" implementation, the difference is just with the returned value.

Additionally it gives more potential values to be generated within the same timeframe. I honestly do not see the point in making that value other than `1`.

Reply:
the previous implementation was a randomly-seeded counter (that resets every millisecond), but this version uses the increased clock precision method (method 3) of section 6.2 of the RFC:

https://www.rfc-editor.org/rfc/rfc9562.html#section-6.2-5.5

I think they are trying to stick within the RFC guidelines for Method 3 and stepping by the same fraction of a ms

Reply:
@ryanwinchester I have no horse in this race, but that option sounds most reasonable. Not having to care about the OS the BEAM is running on is a feature, as are 2 more bits of randomness.